### PR TITLE
ref(outboxes): Add PartnerAccount Update signal

### PR DESCRIPTION
**Pull request by leeandher**
Adds a partner account update signal and changes some docstrings.

### CODE REVIEW

File: src/sentry/models/outbox.py [hunk 1]
```diff
@@ -101,6 +101,8 @@ class OutboxCategory(IntEnum):
     AUTH_PROVIDER_UPDATE = 24
     AUTH_IDENTITY_UPDATE = 25
 
+    PARTNER_ACCOUNT_UPDATE = 26
```
Comment (GabeVillalobos) [inline src/sentry/models/outbox.py:104]: Do we need a new outbox scope for this as well?

**Review by markstory — APPROVED** [review]

File: src/sentry/models/outbox.py (hunk 1 above)
Comment (dashed) [reply src/sentry/models/outbox.py:104]: I placed this in the `ORGANIZATION_SCOPE` based on my skimming of https://github.com/getsentry/getsentry/pull/11630/.

Comment (leeandher) [reply src/sentry/models/outbox.py:104]: I think it'd make sense to throw it under the `PROVISION_SCOPE` since it's usually hit in those flows anyway.

Comment (markstory) [reply src/sentry/models/outbox.py]: :+1: provision makes sense to me, as partner updates have to be sequenced with other provisioning operations.

**Review by markstory — APPROVED** [review]

**Pull request merged**